search/search_integration_tests/search_query_v2_test.cpp

```diff
@@ -255,3 +255,59 @@ UNIT_TEST(SearchQueryV2_Smoke)
     TEST(MatchResults(engine, rules, request.Results()), ());
   }
 }
+
+UNIT_TEST(SearchQueryV2_SearchInWorld)
+{
+  my::ScopedLogLevelChanger const debugLogLevel(LDEBUG);
+
+  classificator::Load();
+  platform::LocalCountryFile testWorld(GetPlatform().WritableDir(),
+                                       platform::CountryFile("testWorld"), 0);
+  auto cleanup = [&]() {Cleanup(testWorld);};
+  cleanup();
+  MY_SCOPE_GUARD(cleanupAtExit, cleanup);
+
+  vector<storage::CountryDef> countries;
+  countries.emplace_back("Wonderland", m2::RectD(m2::PointD(-1.0, -1.0), m2::PointD(1.0, 1.0)));
+
+  TestSearchEngine engine("en", make_unique<storage::CountryInfoGetterForTesting>(countries),
+                          make_unique<TestSearchQueryFactory>());
+
+  auto const wonderland = make_shared<TestCountry>(m2::PointD(0, 0), "Wonderland", "en");
+  auto const losAlamos =
+      make_shared<TestCity>(m2::PointD(0, 0), "Los Alamos", "en", 100 /* rank */);
+
+  {
+    TestMwmBuilder builder(testWorld, feature::DataHeader::world);
+    builder.Add(*wonderland);
+    builder.Add(*losAlamos);
+  }
+
+  auto const result = engine.RegisterMap(testWorld);
+  TEST_EQUAL(result.second, MwmSet::RegResult::Success, ());
+
+  auto worldId = result.first;
+
+  m2::RectD const viewport(m2::PointD(-1.0, -1.0), m2::PointD(-0.5, -0.5));
```

Comment:
mpimenov <notifications@github.com> writes:

> In search/search_integration_tests/search_query_v2_test.cpp:
>
>> +  auto const wonderland = make_shared<TestCountry>(m2::PointD(0, 0), "Wonderland", "en");
>> +  auto const losAlamos =
>> +      make_shared<TestCity>(m2::PointD(0, 0), "Los Alamos", "en", 100 /* rank */);
>> +
>> +  {
>> +    TestMwmBuilder builder(testWorld, feature::DataHeader::world);
>> +    builder.Add(*wonderland);
>> +    builder.Add(*losAlamos);
>> +  }
>> +
>> +  auto const result = engine.RegisterMap(testWorld);
>> +  TEST_EQUAL(result.second, MwmSet::RegResult::Success, ());
>> +
>> +  auto worldId = result.first;
>> +
>> +  m2::RectD const viewport(m2::PointD(-1.0, -1.0), m2::PointD(-0.5, -0.5));
>
> I think there are two common cases:
> 1. Viewport covers everything, i.e. MercatorBounds::FullRect().
> 2. Viewport is completely unrelated to the query.
>
> Should we test both?
>
> —
> Reply to this email directly or view it on GitHub.

I think no. Viewport is unrelated to high-level search, but if you
insist it's not a problem to test both cases.